nats/src/test/kotlin/rys/nats/NatsTestConfiguration.kt

```diff
@@ -0,0 +1,43 @@
+package rys.nats
+
+import com.mongodb.client.MongoClient
+import com.mongodb.client.MongoClients
+import io.nats.client.Connection
+import io.nats.client.Nats
+import io.nats.client.Options
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.SpringBootConfiguration
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
+import org.springframework.context.annotation.Bean
+import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.repository.config.EnableMongoRepositories
+
+@SpringBootConfiguration
+@EnableMongoRepositories(basePackages = ["rys.rest.repository"])
+class NatsTestConfiguration {
+    @Value("\${nats.uri}")
+    private lateinit var natsUri: String
+
+    @Value("\${spring.data.mongodb.uri}")
+    private lateinit var mongoUri: String
+
+    @Bean
+    @ConditionalOnMissingBean
+    fun connection(): Connection {
+        val options = Options.Builder()
+            .server(natsUri)
+            .build()
+        return Nats.connect(options)
+    }
+
+    @Bean
+    fun mongoClient(): MongoClient {
+        return MongoClients.create(mongoUri)
+    }
+
+    @Bean
+    fun mongoTemplate(@Autowired mongoClient: MongoClient): MongoTemplate {
+        return MongoTemplate(mongoClient, "testDB")
+    }
```

Comment:
Do you really need this bean configurations? As I know **MongoTemplate** is configured automatically with some starters. And as for **MongoClient**, I failed to find where you use it. I think that db connect are configured via properties file.